lib/Serialization/Serialization.cpp

```diff
@@ -1819,9 +1835,18 @@ void Serializer::writeCrossReference(const Decl *D) {
   auto val = cast<ValueDecl>(D);
   auto ty = val->getInterfaceType()->getCanonicalType();
   abbrCode = DeclTypeAbbrCodes[XRefValuePathPieceLayout::Code];
+  IdentifierID iid;
+  switch (val->getBaseName().getKind()) {
+  case DeclBaseName::Kind::Normal:
+    iid = addIdentifierRef(val->getBaseName().getIdentifier());
```

Comment:
I didn't do that because of naming ;-)

If it would take `DeclBaseName` it would probably need to be renamed to `addBaseName`. But then, no special name is ever added to the serialised identifiers. But, taking `DeclBaseName` would probably be the right thing to do in retrospect. Naming issue still stands though…